### DISCUSSION TITLE
How to configure such that a certain process is run twice during `nix flake check`?

### DISCUSSION THREAD

**Discussion by shivaraj-bh** [Platonic-Systems/process-compose-flake#78] [Q&A]
See use-case: https://github.com/juspay/services-flake/pull/332#issuecomment-2365467214

**Comment by srid** [comment]
I'm probably misunderstanding the use case here, but can't you just create a wrapper `writeShellApplication` that invokes the original twice, and then use the wrapper script as test?

**Reply by secobarbital** [reply]
The use case is that elasticsearch plugins (actually everything) is shipped with no permissions, even for the owner. We work around that by explicitly giving the owner write permissions to the directories, so that we can remove the files in them in the next run. A more localized way might be to set permissions before use instead of after, so the permission setting is colocated with the file removal.

**Reply by secobarbital** [reply]
Writing the wrapper would work if it is acceptable if we are ok that declaring the elasticsearch service is not going to directly use the original startup script defined in the flake, but instead call it (twice) through this wrapper.
